File: 347-top-k-frequent-elements/top-k-frequent-elements.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number[]}
 */
var topKFrequent = function(nums, k) {
    const map = new Map();
    for(let i = 0; i < nums.length; i++) {
        if (map.has(nums[i]))
            map.set(nums[i], map.get(nums[i]) + 1);
        else
            map.set(nums[i], 1);
    }

    const helper = []
    for(const [key, value] of map) {
        helper.push([key, value])
    }

    helper.sort( (first, second) => { return second[1] - first[1]; });
    ans = [];
    for(let i in helper) {
        ans.push(helper[i][0]);
        if (ans.length === k)
            break;
    }
    return ans;
};

File: 347-top-k-frequent-elements/top-k-frequent-elements.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        vals = {}
        for num in nums:
            if num in vals:
                vals[num] += 1
            else:
                vals[num] = 1
        print(vals)
        pairs = []
        for val in vals:
            pairs.append([vals[val], val])
        pairs.sort(reverse=True)
        print(pairs)
        ans = []
        for pair in pairs:
            if len(ans) == k:
                return ans
            ans.append(pair[1])
        return ans